nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -288,7 +290,7 @@ public void process(InputStream in) throws IOException {
         int fragmentIndex = 1; // set to 1 to preserve the existing behavior *only*. Perhaps should be deprecated to follow the 0,1,2... scheme
         String fragmentId = UUID.randomUUID().toString();
 
-        if (computedSplitsInfo.size() == 0) {
+        if ((computedSplitsInfo.size() == 0) && (headerFlowFile != null)) {
```

Comment:
Thanks @mattyb149 The part of processor description is added since ver 1.1.0 by NIFI-2851. So I thought @olegz has some reason for that. As I mentioned within the previous comment, ver 1.0.0 and earlier don't create that header only flow files.

With a situation that you mentioned, I'd like to have no output from SplitText to split relationship. If it outputs empty flow files but only with header, I'd have to filter out those files in downstream flow.

Maybe this should be discussed separately from this PR. Since this PR fixes its target issue, I'll merge this into master.